src/main/rx/sumd.c

```diff
@@ -54,60 +54,70 @@
 #define SUMD_BUFFSIZE (SUMD_MAX_CHANNEL * 2 + 5) // 6 channels + 5 = 17 bytes for 6 channels
 
 #define SUMD_BAUDRATE 115200
+#define SUMD_TIME_NEEDED_PER_FRAME 4000
+
+#define SUMD_OFFSET_CHANNEL_1_HIGH 3
+#define SUMD_OFFSET_CHANNEL_1_LOW 4
+#define SUMD_BYTES_PER_CHANNEL 2
+#define SUMD_SYNC_BYTE_INDEX 0
+#define SUMD_CHANNEL_COUNT_INDEX 2
+
+#define SUMD_HEADER_LENGTH 3
+#define SUMD_CRC_LENGTH 2
+
+#define SUMDV1_FRAME_STATE_OK 0x01
+#define SUMDV3_FRAME_STATE_OK 0x03
+#define SUMD_FRAME_STATE_FAILSAFE 0x81
 
 static bool sumdFrameDone = false;
 static uint16_t sumdChannels[MAX_SUPPORTED_RC_CHANNEL_COUNT];
 static uint16_t crc;
 
 static uint8_t sumd[SUMD_BUFFSIZE] = { 0, };
 static uint8_t sumdChannelCount;
+static timeDelta_t lastFrameDelta = 0;
 
 // Receive ISR callback
 static void sumdDataReceive(uint16_t c, void *data)
 {
     UNUSED(data);
 
-    uint32_t sumdTime;
-    static uint32_t sumdTimeLast;
+    static timeUs_t sumdTimeLast;
     static uint8_t sumdIndex;
+    static timeUs_t lastFrameCompleteTimeUs = 0;
 
-    sumdTime = micros();
-    if ((sumdTime - sumdTimeLast) > 4000)
+    const timeUs_t sumdTime = micros();
```

Comment:
This could potentially be changed to `microsISR()` since we know it is called within an ISR.

Reply:
What's the advantage or risk? Haven't looked into that area yet. All the other protocols also use normal `micros()`.

Reply:
`micros()` does an internal check for elevated priority to detect if it's called from within an ISR, and yields to `microsISR()` if so. This is a handy protection for when `micros()` is called in functions that may or may not be called from within an ISR. I think in cases where we want to get microseconds directly inside an ISR calling `microsISR()` is more explicit, and helps to keep the ISR / time we are running with elevated priority at a minimum.
Alternatively we should look into making `microsISR()` static and replacing everywhere to be consistent.

Reply:
Updated to use `microsISR()` in the interrupt handler.